### DISCUSSION TITLE
Distorted rendering in GUI app

### DISCUSSION THREAD

**Discussion by 573** [nix-community/NixOS-WSL#355] [Q&A] [answered]
When I'm using the gtk app simple-scan the rendering is distorted, is there a way to workaround this ?

![grafik](https://github.com/nix-community/NixOS-WSL/assets/123878/f6c33644-b6f9-43e6-b2cd-244fdb1c15ed)

**Comment by nzbr** [comment]
I can, unfortunately not reproduce this behavior. Neither on NixOS nor Ubuntu, so it's probably a bug in WSL itself. can you try if it works on another distro?

**Comment by 573** [comment]
Probably have the wrong WSL version installed:
>❯ wsl.exe -v
WSL-Version: 2.0.9.0
Kernelversion: 5.15.133.1-1
WSLg-Version: 1.0.59
MSRDC-Version: 1.2.4677
Direct3D-Version: 1.611.1-81528511
DXCore-Version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows-Version: 10.0.19045.3570

>❯ nixos-wsl-version
NixOS-WSL DEV_BUILD 5ee4fa3515de7b5609e6d161b800d91328a7a143 native

**Reply by nzbr** [reply]
The WSL version looks fine. I currently have `2.0.14.0`, so updating _might_ be worth tring anyway

**Reply by 573** [reply, marked as answer]
[Updating](https://github.com/microsoft/WSL/releases/tag/2.1.1) removed the issue for me:

>wsl --version
WSL-Version: 2.1.1.0
Kernelversion: 5.15.146.1-2
WSLg-Version: 1.0.60
MSRDC-Version: 1.2.5105
Direct3D-Version: 1.611.1-81528511
DXCore-Version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows-Version: 10.0.19045.3930

Also, just for the record I happen to have this in [`.wslgconfig`](https://github.com/microsoft/wslg/wiki/WSLg-Configuration-Options-for-Debugging), don't know if still needed - might interfer with your mileage:
>[system-distro-env]
;disable GPU in system-distro
LIBGL_ALWAYS_SOFTWARE=1

And BTW I happened to update (see `dxdiag` for the card on your system) the NVIDIA driver beforehand to the newest.
